react-interactive/src/ProgressBar.tsx

```diff
@@ -64,6 +65,11 @@ function ProgressBar(props: IProps) {
 
     }, [props.activeStep, props.steps])
 
+    const clickhandle = React.useCallback((step: number|string) => {
+        if (props.onClick !== undefined)
+            props.onClick(props.activeStep,step);
+    },[props.activeStep]);
```

Comment:
Should this also update when props.onClick does?

Reply:
I considered that but then we have issues if `props.OnClick` is defined as arrow function (e.g. `() => {}`) because it will update on every render

Reply:
I suppose that wouldn't be a huge issue. We probably should not be doing that anyway.... a re-render should be fairly quick and realistically we should be wrapping all functions in `React.callback` in the parent anyway....

Reply:
```suggestion
    },[props.activeStep, props.onClick]);
```